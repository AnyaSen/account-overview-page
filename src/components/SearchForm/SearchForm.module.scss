@import "../../assets/scss/variables";
@import "../../assets/scss/breakpoints";

.SearchForm {
  display: flex;
  flex-direction: column;
  text-align: center;
  margin-top: $space-1;
  padding-bottom: $space-2;
  flex-wrap: wrap;
  border-bottom: 1px solid $color-border;

  h1 {
    margin: $space-2;
  }

  @include sm {
    padding-bottom: $space-1;
    margin: 0;

    h1 {
      margin: $space-0;
    }
  }
}

.SearchFormContainer {
  display: flex;
  justify-content: center;

  @include md {
    flex-direction: column;
  }
}

.inputsContainer {
  margin: $space-0 $space-2;

  @include sm {
    margin: 0.7rem;
  }
}

.btnsContainer {
  display: flex;
  align-items: center;

  @include md {
    justify-content: center;
  }
}
