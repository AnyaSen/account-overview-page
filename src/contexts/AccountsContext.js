import React, { useState, createContext } from "react";

export const AccountsContext = createContext();

export const AccountsContextProvider = ({ children }) => {
  const [accNumber, setAccNumber] = useState(90);
  const [accName, setAccName] = useState("FUNDING_ACCOUNT");
  const [accID, setAccID] = useState("DKUW67");
  const [accBalance, setAccBalance] = useState("60â‚¬");

  return (
    <AccountsContext.Provider
      value={{
        accNumber,
        accName,
        accID,
        accBalance
      }}
    >
      {children}
    </AccountsContext.Provider>
  );
};
