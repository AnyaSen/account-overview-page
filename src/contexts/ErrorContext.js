import React, { createContext, useState } from "react";

export const ErrorContext = createContext();

export const ErrorContextProvider = ({ children }) => {
  const [isAccountsError, setIsAccountsError] = useState(false);
  const [isUserInfoError, setIsUserInfoError] = useState(false);

  return (
    <ErrorContext.Provider
      value={{
        isAccountsError,
        setIsAccountsError,
        isUserInfoError,
        setIsUserInfoError
      }}
    >
      {children}
    </ErrorContext.Provider>
  );
};
