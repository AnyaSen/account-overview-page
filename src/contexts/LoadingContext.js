import React, { createContext, useState } from "react";

export const LoadingContext = createContext();

export const LoadingContextProvider = ({ children }) => {
  const [isAccountsLoading, setIsAccountsLoading] = useState(true);
  const [isUserInfoLoading, setIsUserInfoLoading] = useState(true);

  return (
    <LoadingContext.Provider
      value={{
        isAccountsLoading,
        setIsAccountsLoading,
        isUserInfoLoading,
        setIsUserInfoLoading
      }}
    >
      {children}
    </LoadingContext.Provider>
  );
};
