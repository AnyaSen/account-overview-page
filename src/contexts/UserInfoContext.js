import React, { useState, createContext } from "react";

export const UserInfoContext = createContext();

export const UserInfoContextProvider = ({ children }) => {
  const [name, setName] = useState("SOME NAME");

  return (
    <UserInfoContext.Provider
      value={{
        name
      }}
    >
      {children}
    </UserInfoContext.Provider>
  );
};
