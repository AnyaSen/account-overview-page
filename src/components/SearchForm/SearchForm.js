import React, { useContext, useState, useEffect } from "react";
import Styles from "./SearchForm.module.scss";

import InputField from "./InputField/InputField";
import PrimaryButton from "../Buttons/PrimaryButton/PrimaryButton";
import SecondaryButton from "../Buttons/SecondaryButton/SecondaryButton";

import { AccountsContext } from "../../contexts/AccountsContext";

import {
  filterAccountsByID,
  filterAccountsByName,
  filterAccountsByNameAndID
} from "../../services/filterAccounts";

export default function SearchForm() {
  const { modifiedAccounts, setModifiedAccounts, accounts } = useContext(
    AccountsContext
  );

  const [nameValue, setNameValue] = useState("");
  const [idValue, setIdValue] = useState("");

  const [isDisabledSearchBtn, setIsDisabledSearchBtn] = useState(true);
  const [isDisabledShowAllBtn, setIsDisabledShowAllBtn] = useState(true);

  useEffect(() => {
    const inputsAreNotEmpty = nameValue || idValue;

    setIsDisabledSearchBtn(!inputsAreNotEmpty);
  }, [nameValue, idValue]);

  const onChangeNameValue = event => {
    setNameValue(event.target.value);
  };

  const onChangeIdValue = event => {
    setIdValue(event.target.value);
  };

  const handleSubmit = () => {
    const filteredAccountsByName = filterAccountsByName(
      modifiedAccounts,
      nameValue
    );
    const filteredAccountsById = filterAccountsByID(modifiedAccounts, idValue);

    const filteredAccountsByNameAndID = filterAccountsByNameAndID(
      nameValue,
      idValue,
      modifiedAccounts
    );

    if (!nameValue) {
      setModifiedAccounts(filteredAccountsById);
    } else if (!idValue) {
      setModifiedAccounts(filteredAccountsByName);
    } else if ((nameValue, idValue)) {
      setModifiedAccounts(filteredAccountsByNameAndID);
    }

    setNameValue("");
    setIdValue("");

    setIsDisabledShowAllBtn(false);
  };

  const showAll = () => {
    setModifiedAccounts(accounts);
    setIsDisabledShowAllBtn(true);
  };

  return (
    <div data-testid="Search-form" className={Styles.SearchForm}>
      <h1>INVESTMENT ACCOUNTS OVERVIEW</h1>

      <form
        onSubmit={e => e.preventDefault()}
        className={Styles.SearchFormContainer}
      >
        <div className={Styles.inputsContainer}>
          <InputField
            placeholder="Name"
            inputValue={nameValue}
            onChange={onChangeNameValue}
          />

          <InputField
            placeholder="ID"
            inputValue={idValue}
            onChange={onChangeIdValue}
          />
        </div>
        <div className={Styles.btnsContainer}>
          <PrimaryButton
            btnText="Search"
            isDisabled={isDisabledSearchBtn}
            onClick={handleSubmit}
          />

          <SecondaryButton
            btnText="Show all"
            isDisabled={isDisabledShowAllBtn}
            onClick={showAll}
          />
        </div>
      </form>
    </div>
  );
}
