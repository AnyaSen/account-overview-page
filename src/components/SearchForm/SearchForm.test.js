import React from "react";
import { render, cleanup } from "@testing-library/react";

import { AccountsContextProvider } from "../../contexts/AccountsContext";
import { LoadingContextProvider } from "../../contexts/LoadingContext";
import { ErrorContextProvider } from "../../contexts/ErrorContext";

import SearchForm from "./SearchForm";

describe("<SearchForm />", () => {
  afterAll(cleanup);

  const createRenderTree = () => (
    <ErrorContextProvider>
      <LoadingContextProvider>
        <AccountsContextProvider>
          <SearchForm />
        </AccountsContextProvider>
      </LoadingContextProvider>
    </ErrorContextProvider>
  );

  test("Should have have 2 children.", () => {
    const tree = createRenderTree();

    const { getByTestId, container } = render(tree);

    const SearchFormContainer = getByTestId("Search-form");

    expect(SearchFormContainer.children.length).toEqual(2);

    expect(container.firstChild).toMatchSnapshot();
  });
});
