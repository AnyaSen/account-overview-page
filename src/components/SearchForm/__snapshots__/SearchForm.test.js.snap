// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchForm /> Should have have 2 children. 1`] = `
<div
  class="SearchForm"
  data-testid="Search-form"
>
  <h1>
    INVESTMENT ACCOUNTS OVERVIEW
  </h1>
  <form
    class="SearchFormContainer"
  >
    <div
      class="inputsContainer"
    >
      <input
        class="InputField"
        data-testid="InputField"
        placeholder="Name"
        type="text"
        value=""
      />
      <input
        class="InputField"
        data-testid="InputField"
        placeholder="ID"
        type="text"
        value=""
      />
    </div>
    <div
      class="btnsContainer"
    >
      <button
        class="PrimaryButton"
        disabled=""
        type="submit"
        value="submit"
      >
        Search
      </button>
      <button
        class="SecondaryButton"
        disabled=""
      >
        Show all
      </button>
    </div>
  </form>
</div>
`;
